pkg/shipper/shipper_test.go

```diff
@@ -0,0 +1,116 @@
+package shipper
+
+import (
+	"testing"
+
+	"context"
+	"io/ioutil"
+	"time"
+
+	"fmt"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/improbable-eng/promlts/pkg/testutil"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/tsdb/labels"
+)
+
+type inMemStorage struct {
+	t    *testing.T
+	dirs map[string]struct{}
+}
+
+func newInMemStorage(t *testing.T) *inMemStorage {
+	return &inMemStorage{
+		t:    t,
+		dirs: make(map[string]struct{}),
+	}
+}
+
+func (r *inMemStorage) Exists(_ context.Context, id string) (bool, error) {
+	_, exists := r.dirs[id]
+	return exists, nil
+}
+
+func (r *inMemStorage) Upload(_ context.Context, dir string) error {
+	// Double check if shipper checks Exists method properly.
+	_, exists := r.dirs[dir]
+	testutil.Assert(r.t, !exists, "target should not exists")
+
+	r.dirs[dir] = struct{}{}
+	return nil
+}
+
+func TestShipper_UploadsBlocksFromProm(t *testing.T) {
+	logger := log.NewNopLogger()
+
+	dir, err := ioutil.TempDir("", "shipper-test-snapshots")
+	testutil.Ok(t, err)
+	defer os.RemoveAll(dir)
+
+	p, err := testutil.NewTSDB()
+	testutil.Ok(t, err)
+	defer p.Close()
+
+	storage := newInMemStorage(t)
+	shipper := New(
+		logger,
+		nil,
+		dir,
+		storage,
+		IsULIDDir,
+	)
+
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
`context.Background()` is fine here. Per docs, TODO is only for places where you want to connect it later with an incoming context.

Reply:
when I was list visiting it... there was `...and tests` comment. (: Either my memory is weak or they changed that recently.

Reply:
Doesn't seem so anymore: https://godoc.org/context#TODO